File: SurveyAPI/SurveyAPI/Models/QuestionStats.cs
namespace SurveyAPI.Models {
    public class QuestionStats {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool Multiple { get; set; } = false;
        public List<ChoiceStats> Choices { get; set; } = new List<ChoiceStats>();
    }
}

File: front-vue/src/models/question/QuestionStats.ts
import { ChoiceStats, ChoiceStatsDTO } from "./ChoiceStats";

export class QuestionStatsDTO {
  id = "";
  name = "";
  beginDate = "";
  endDate = "";
  multiple = false;
  choices: ChoiceStatsDTO[] = [];
}

export class QuestionStats extends QuestionStatsDTO {
  constructor(data: QuestionStats | QuestionStatsDTO | null) {
    super();
    if (data) {
      Object.assign(this, data);
      this.choices = data.choices.map((c) => new ChoiceStats(c));
    }
  }
}
